frontend/ui/network/manager.lua

```diff
@@ -124,7 +124,7 @@ function NetworkMgr:getWifiMenuTable()
         enabled_func = function() return Device:isAndroid() or Device:isCervantes() or Device:isKindle() or Device:isKobo() or Device:isSonyPRSTUX() end,
         checked_func = function() return NetworkMgr:isWifiOn() end,
         callback = function(touchmenu_instance)
-            local wifi_status = NetworkMgr:isWifiOn()
+            local wifi_status = NetworkMgr:isWifiOn() and NetworkMgr:isConnected()
```

Comment:
You might want to git blame check here since I think this line has had some weirdness associated with it.